### DISCUSSION TITLE
Question on redeliver option

### DISCUSSION THREAD

**Discussion by killmasta93** [postalserver/postal#3114] [Help with using Postal]
Hi
Currently was wondering if someone is wrong with my setup as currently i put a limit every 60min 250 emails but i have around 3k which has passed more then 72 hours in the held part, 
I was wondering how long it stays in held before going to queue?
or is there a way to manually or automatic to run to release those emails?

Thanks

**Comment by willpower232** [comment]
Held emails will never be released automatically. If you look around the discussions, I think someone had some code for releasing all held messages. Alternatively you could open each message in a tab and click the Release Message button.

**Reply by killmasta93** [reply]
Thank you yeah running this script 

```

server_id = 1  # Replace with your actual server ID
db = Server.find(server_id).message_db

# Fetch all held messages in one go
query = { where: { held: 1 } }
messages = db.messages(query)

# Process each message and release it
messages.each do |m|
  m.add_to_message_queue(manual: true)
end

puts "All held messages have been released."


```
out of curiosity how can i delete the messages which are in the queue,?

Thanks
